### ISSUE TITLE
Generate where all of them implements Send + Sync?

### ISSUE THREAD

**Issue by beckend** [blazzy/podman-rest-client#1]
So we can store and share this in an Arc<Client> and reuse and not need to connect all the time.

**Comment by beckend** [comment]
```shell
note: future is not `Send` as this value is used across an await
   --> packages/podman_cli/src/modules/podman/deploy/handle_pods.rs:34:57
    |
25  |   let api_pods = client.pods_api();
    |       -------- has type `&dyn podman_autogen_api::apis::pods_api::PodsApi` which is not `Send`
```

**Comment by beckend** [comment]
I solved it by using `client.pods_api()` every time I needed the sub APIs.

**Comment by blazzy** [comment]
@backend  Ran into the same annoyance. I submitted a patch to the upstream openapi-generator project here: https://github.com/OpenAPITools/openapi-generator/pull/1920

But I can push out a new version of this library before that's accepted upstream. Making everything `Send` (and probably `Sync` too)

**Comment by beckend** [comment]
Actually, I tried pulling an image, creating a container. Both of the APIs failed.

**Comment by blazzy** [comment]
@beckend

The latest version published makes all those API structs `Send + Sync` now.

>  Actually, I tried pulling an image, creating a container. Both of the APIs failed.

Which version of podman are you using? This crate was built around v5.

I just wrote some tests and some of them are currently failing on github because the github ubuntu runner is on v4.

**Comment by beckend** [comment]
5.1.0

**Comment by blazzy** [comment]
@beckend

Linux or Mac? One thing that trips me up on a new linux system is remembering to enable the unix socket. Since podman doesn't have a server by default like docker. I added some documentation [here](https://github.com/blazzy/podman-rest-client?tab=readme-ov-file#podman-socket)  about that.

Are you using the `guess_configuration` function? One small breaking change is that it has been moved to be `Config::guess` now. The function now returns an error if it can't find a socket. Before it would silently let you go ahead and then error later when you tried to make a request.

Now that I realize people are using this already, I started a [changelog](https://github.com/blazzy/podman-rest-client?tab=readme-ov-file#changelog) and I'm actually tagging releases on github.
